package-res/lib/protovis.js

```diff
@@ -3670,9 +3670,11 @@ pv.Scale.quantitative = function() {
 
   /** @private */
   function scale(x) {
+    if (x >= Math.pow(2,53)) { x = x / Math.pow(2,53); }
```

Comment:
I believe clamping would be more appropriate. Also, you should take negative overflow into account.

Please, store the value of Math.pow(2,53) as a constant somewhere in the module scope:
```js
var MAX_SAFE_INTEGER = Math.pow(2,53) - 1;
```

Then clamp it:
```js
x = Math.max(-MAX_SAFE_INTEGER, Math.min(x, MAX_SAFE_INTEGER));
```